plugins/base/src/main/kotlin/translators/documentables/DefaultPageCreator.kt

```diff
@@ -611,6 +630,19 @@ internal val Documentable.customTags: Map<String, SourceSetDependent<CustomTagWr
 private val Documentable.hasSeparatePage: Boolean
     get() = this !is DTypeAlias
 
+/**
+ * @see DefaultPageCreator.sortDivergentElementsDeterministically for usage
+ */
+private val divergentDocumentableComparator =
+    compareBy<Documentable, String?>(nullsLast()) { it.dri.packageName }
+        .thenBy(nullsFirst()) { it.dri.classNames } // nullsFirst for top level to be first
```

Comment:
I would like to have the function `topLevelExtensionFirst` instead of the comment.

Reply:
Like this?

```kotlin
fun Comparator<Documentable>.topLevelExtensionFirst() : Comparator<Documentable> = 
    thenBy(nullsFirst()) { it.dri.classNames }

compareBy<Documentable, String?>(nullsLast()) { it.dri.packageName }
    .topLevelExtensionFirst()
    .thenBy(..)
```

If yes, why?

Reply:
Something like:

```kt
fun Comparator<Documentable>.thenTopLevelExtensionFirst(comparatorOfClassNames: Comparator<String>) = thenBy (nullsFirst(comparatorOfClassNames)) { it.dri.classNames }
private val divergentDocumentableComparator =
    compareBy<Documentable, String?>(nullsLast()) { it.dri.packageName }
    .thenTopLevelExtensionFirst(comparatorOfClassNames = naturalOrder())
```
It needs no code comment.

Reply:
Sorry, but in this particular case I don't think having a separate function makes it any more readable - if anything, it makes it a bit more complicated than it needs to be. I don't think the function will be re-usable or that it encapsulates some logic that will be easier to refactor/add to.

Note on the naming: it's a comparison of divergent elements, and types and members are considered to be that as well, so `thenTopLevelExtensionFirst` would have to be renamed to something like `thenByClassNameWithTopLevelMemersFirst`

Reply:
Up to you